telegram/telegram.go

```diff
@@ -0,0 +1,740 @@
+	btnSetFilters     = userMenu.Text("ثبت فیلتر")
+	btnShareBookmarks = userMenu.Text("اشتراک گذاری آگهی های مورد علاقه")
+	btnGetOutputFile  = userMenu.Text("خروجی گرفتن از آگهی ها")
+	btnDeleteHistory  = userMenu.Text("پاک کردن تاریخچه")
+	// تنظیم بازه زمانی
+
+	filterMenu        = &telebot.ReplyMarkup{ResizeKeyboard: true}
+	btnPrice          = filterMenu.Text("قیمت")
+	btnCity           = filterMenu.Text("شهر")
+	btnNeighborhood   = filterMenu.Text("محله")
+	btnArea           = filterMenu.Text("متراژ")
+	btnNumberOfRooms  = filterMenu.Text("تعداد اتاق خواب")
+	btnAge            = filterMenu.Text("سن بنا")
+	btnCategoryAV     = filterMenu.Text("ویلایی یا آپارتمانی؟")
+	btnFloorNumber    = filterMenu.Text("طبقه")
+	btnStorage        = filterMenu.Text("انباری")
+	btnElevator       = filterMenu.Text("آسانسور")
+	btnAdDate         = filterMenu.Text("تاریخ ایجاد آگهی")
+	btnSendFilter     = filterMenu.Text("ثبت فیلتر")
+	btnBackFilterMenu = filterMenu.Text("بازگشت")
+
+	purchaseOrRentMenu = &telebot.ReplyMarkup{ResizeKeyboard: true}
+	btnPurchase        = purchaseOrRentMenu.Text("خرید")
+	btnRent            = purchaseOrRentMenu.Text("اجاره")
+
+	getOutputFileMenu        = &telebot.ReplyMarkup{ResizeKeyboard: true}
+	btnGetAsZip              = getOutputFileMenu.Text("دریافت آگهی ها به صورت فایل زیپ")
+	btnGetViaEmail           = getOutputFileMenu.Text("دریافت آگهی ها از طریق ایمیل")
+	btnBackGetOutputFileMenu = getOutputFileMenu.Text("بازگشت")
+)
+
+func NewTelegramBot(config *TelegramConfig) (*Telegram, error) {
+	pref := telebot.Settings{
+		Token:  config.Token,
+		Poller: &telebot.LongPoller{Timeout: 10 * time.Second},
+	}
+
+	bot, err := telebot.NewBot(pref)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create bot: %w", err)
+	}
+
+	logFile, err := os.Create("log/telegram.log")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create telegram log file: %w", err)
+	}
+
+	infoLogger := log.New(logFile, "INFO: ", log.LstdFlags)
+	errorLogger := log.New(logFile, "ERROR: ", log.LstdFlags)
+
+	telegramLogger := logg.TelegramLogger{
+		InfoLogger:  infoLogger,
+		ErrorLogger: errorLogger,
+	}
+
+	telegram := &Telegram{
+		Bot:     bot,
+		Config:  config,
+		Loggers: telegramLogger,
+	}
+
+	return telegram, nil
+}
+
+func (t *Telegram) registerHandlers() {
+	t.Bot.Handle("/start", t.handleStart)
+	t.Bot.Handle(telebot.OnText, t.handleText)
+}
+
+func (t *Telegram) Start() {
+	t.registerHandlers()
+	t.Loggers.InfoLogger.Println("Starting Telegram bot...")
+	t.Bot.Start()
+}
+
+func (t *Telegram) handleStart(c telebot.Context) (err error) {
+	welcomeMsg := "به ربات خزنده خوش اومدین :)"
+
+	telegram_ID := c.Sender().ID
+	gormUser := repository.NewGormUser(database.GetInstnace().Db)
+	user, e := gormUser.GetByTelegramId(strconv.Itoa(int(telegram_ID)))
+	if e != nil && errors.Is(e, gorm.ErrRecordNotFound) {
+		user, _ = gormUser.Add(models.Users{TelegramId: strconv.Itoa(int(telegram_ID)), Role: "user"})
+	}
+	if user.Role == "user" {
```

Comment:
It's more effective to use a variable name instead of "user" to enhance readability.